runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java

```diff
@@ -677,6 +680,40 @@ private void processInventoryLoot(String event, LootRecordType lootRecordType, I
 		}
 	}
 
+	private boolean processHerbSackLoot(int timestamp, String event, LootRecordType lootRecordType)
+	{
+		List<ItemStack> herbs = new ArrayList<>();
+		for (Object messageObject : client.getMessages())
+		{
+			MessageNode messageNode = (MessageNode) messageObject;
+			if (messageNode.getTimestamp() == timestamp && messageNode.getType() == ChatMessageType.SPAM)
+			{
+				String herbName;
+				if (messageNode.getValue().startsWith("You put the Grimy"))
+				{
+					herbName = messageNode.getValue().replace("You put the ", "").replace(" herb into your herb sack.", "");
+					herbs.add(new ItemStack(itemManager.search(herbName).get(0).getId(), 1, client.getLocalPlayer().getLocalLocation()));
+				}
+				// Edge case for full herb sack for some of the looted herbs
+				else if (messageNode.getValue().startsWith("Your herb sack is too full to hold the Grimy"))
+				{
+					herbName = messageNode.getValue().replace("Your herb sack is too full to hold the ", "").replace(" herb.", "");
+					herbs.add(new ItemStack(itemManager.search(herbName).get(0).getId(), 1, client.getLocalPlayer().getLocalLocation()));
+				}
+			}
+		}
+
+		if (herbs.isEmpty())
+		{
+			return false;
+		}
+		else
```

Comment:
Since the `if` branch returns from this method, the rest of this code does not need to be in an `else` block. That is, this should be replaced by the following:

```java
if (herbs.isEmpty())
{
	return false;
}

addLoot(event, -1, lootRecordType, herbs);
return true;
```